Adding more endpoints around releases and release assets.

tmcgilchrist opened the pull request:
New endpoints for:
 * Get the latest release (https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#get-the-latest-release)
 * Get a release asset (https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#get-a-release-asset)
 * Delete a release asset (https://docs.github.com/rest/reference/repos#delete-a-release-asset)
 * Delete a release asset (https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#delete-a-release-asset)
 * List assets in a release (https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#list-release-assets)
 
Updated endpoint for:
 * Get release by tag name - previously this was getting all releases and finding the matching tag, there is a new direct API for doing this. (https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#get-a-release-by-tag-name)

I'm still working on adding other release based endpoints that we don't currently support so I'll update this as I go.

Aaylor commented on lib/github_core.ml:
```diff
@@ -1617,6 +1618,16 @@ module Make(Env : Github_s.Env)(Time : Github_s.Time)(CL : Cohttp_lwt.S.Client)
       API.post ?token ~params ~headers ~body ~uri ~expected_code:`Created
         (fun _b -> return ())
+
+    (* let delete_asset Delete a release asset
```
These comments are not necessary here ! ;)
If they are not listed in the `Not yet supported` part of the `README.md`, you can add them here!

tmcgilchrist replied:
No worries, I've added that function now.

Aaylor commented:
LGTM, minus comments & the conflict! :)

(Ignore the failure of `ocaml-ci`, I'll fix this later, this is due to the documentation)

tmcgilchrist commented:
Fixed up the conflicts and added a few extra endpoints. Still need to write a nice test for exercising the new release endpoints. Thanks for the review @Aaylor

avsm commented:
Now you're working on this repository again, can this PR be rebased @tmcgilchrist? (no rush!)

tmcgilchrist commented:
```
github.opam: changes needed:
  "github_data" {>= 0}                     [from lib]
Note: version numbers are just suggestions based on the currently installed version.
Run with -f to apply changes in non-interactive mode.
"/bin/bash" "-c" "opam exec -- opam-dune-lint" failed with exit status 1
2021-06-23 02:45.32: Job failed: Failed: Build failed
2021-06-23 02:45.32: Log analysis:
2021-06-23 02:45.32: >>> 
github.opam: changes needed: (score = 30)
2021-06-23 02:45.32: Lint failed for github.opam
```

talex5 commented:
The opam file needs to use the opam package name, not the internal dune name. The internal name works for a local dev build when you're building both together, but not when the packages are released.
